vm-setup/roles/v1aX_integration_test/tasks/remediation.yml

```diff
@@ -103,39 +117,42 @@
     register: shutdown_vms
     retries: 50
     delay: 10
-    until: WORKER_NODE_VM in shutdown_vms.list_vms
+    until: MASTER_NODE_1_VM and MASTER_NODE_2_VM in shutdown_vms.list_vms
```

Comment:
I don't think this works this way. It will check that MASTER_NODE_2_VM is in the list, but it will just check that MASTER_NODE_1_VM is defined. I think you need the "in" on both sides of the "and"

Reply:
I have tested locally and it works. 
```
TASK [v1aX_integration_test : List only running VMs] **************************************************************************************************************************************************************
task path: /home/ubuntu/metal3-dev-env/vm-setup/roles/v1aX_integration_test/tasks/test.yml:147
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: ubuntu
<localhost> EXEC /bin/sh -c 'echo ~ubuntu && sleep 0'
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/ubuntu/.ansible/tmp/ansible-tmp-1592382887.4515939-50949775503133 `" && echo ansible-tmp-1592382887.4515939-50949775503133="` echo /home/ubuntu/.ansible/tmp/ansible-tmp-1592382887.4515939-50949775503133 `" ) && sleep 0'
Using module file /usr/local/lib/python3.6/dist-packages/ansible/modules/cloud/misc/virt.py
<localhost> PUT /home/ubuntu/.ansible/tmp/ansible-local-20549pmglqeh2/tmpg5nwtfdx TO /home/ubuntu/.ansible/tmp/ansible-tmp-1592382887.4515939-50949775503133/AnsiballZ_virt.py
<localhost> EXEC /bin/sh -c 'chmod u+x /home/ubuntu/.ansible/tmp/ansible-tmp-1592382887.4515939-50949775503133/ /home/ubuntu/.ansible/tmp/ansible-tmp-1592382887.4515939-50949775503133/AnsiballZ_virt.py && sleep 0'
<localhost> EXEC /bin/sh -c 'sudo -H -S -n  -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-xpqwqqwgzbnaamunklqndxarzwatkvjy ; /usr/bin/python /home/ubuntu/.ansible/tmp/ansible-tmp-1592382887.4515939-50949775503133/AnsiballZ_virt.py'"'"' && sleep 0'
<localhost> EXEC /bin/sh -c 'rm -f -r /home/ubuntu/.ansible/tmp/ansible-tmp-1592382887.4515939-50949775503133/ > /dev/null 2>&1 && sleep 0'
ok: [localhost] => {
    "attempts": 1,
    "changed": false,
    "invocation": {
        "module_args": {
            "autostart": null,
            "command": "list_vms",
            "name": null,
            "state": "running",
            "uri": "qemu:///system",
            "xml": null
        }
    },
    "list_vms": [
        "node_0",
        "node_1",
        "node_3",
        "node_2"
    ]
}

```

Reply:
I can try to test by adding "in".

Reply:
it failed unfortunately when I tested with:
```
  - name: List only powered off VMs
    virt:
      command: list_vms
      state: shutdown
    register: shutdown_vms
    retries: 50
    delay: 10
    until: MASTER_NODE_1_VM in and MASTER_NODE_2_VM in shutdown_vms.list_vms
    become: yes
```

and
```
  - name: List only powered off VMs
    virt:
      command: list_vms
      state: shutdown
    register: shutdown_vms
    retries: 50
    delay: 10
    until: MASTER_NODE_1_VM in and in MASTER_NODE_2_VM shutdown_vms.list_vms
    become: yes
```

```
<localhost> EXEC /bin/sh -c 'rm -f -r /home/ubuntu/.ansible/tmp/ansible-tmp-1592383828.358186-151715696438101/ > /dev/null 2>&1 && sleep 0'
fatal: [localhost]: FAILED! => {
    "msg": "The conditional check 'MASTER_NODE_1_VM in and MASTER_NODE_2_VM in shutdown_vms.list_vms' failed. The error was: template error while templating string: expected token 'end of statement block', got 'MASTER_NODE_2_VM'. String: {% if MASTER_NODE_1_VM in and MASTER_NODE_2_VM in shutdown_vms.list_vms %} True {% else %} False {% endif %}"
}

PLAY RECAP ********************************************************************************************************************************************************************************************************
localhost  
```
```
fatal: [localhost]: FAILED! => {
    "msg": "The conditional check 'MASTER_NODE_1_VM in and in MASTER_NODE_2_VM shutdown_vms.list_vms' failed. The error was: template error while templating string: expected token 'end of statement block', got 'shutdown_vms'. String: {% if MASTER_NODE_1_VM in and in MASTER_NODE_2_VM shutdown_vms.list_vms %} True {% else %} False {% endif %}"
}

PLAY RECAP ********************************************************************************************************************************************************************************************************
localhost                  : ok=17   changed=9    unreachable=0    failed=1    skipped=37   rescued=0    ignored=0   

```

Reply:
I meant MASTER_NODE_1_VM in shutdown_vms.list_vms and MASTER_NODE_2_VM in shutdown_vms.list_vms

Reply:
Thanks @maelk . I'll fix that.

Reply:
Done.